application/src/main/webapp/js/chat.js

```diff
@@ -1648,17 +1659,41 @@ ChatApplication.prototype.getUsers = function(roomId, callback, asString) {
  * @param filter : the filter (ex: Ben Pa)
  * @param callback : return the json users data list as a parameter of the callback function
  */
-ChatApplication.prototype.getAllUsers = function(filter, callback) {
+ChatApplication.prototype.getAllUsers = function(filter, isSearchInRoomFirst, callback) {
+
+  // Cache result
+  var _this = this;
+  if (this.getAllUsersCache == undefined) {
+    this.getAllUsersCache = {};
+  }
+  var cacheKey = filter;
+  if (isSearchInRoomFirst) {
+    cacheKey += '__' + this.targetUser;
+  }
+  if (this.getAllUsersCache[cacheKey] != undefined) {
```

Comment:
Adding a chat here could be a good idea, but it seems it is probably too basic. Once the cache has been populated, it is never invalidated, so if a new user has been created or created in the meanwhile, the cache will not be updated. The only way to reload the cache is to refresh the page.
Always, there is no max capacity in this cache.